### PR TITLE
Flesh out implementation for TryNode wrapping a streaming node

### PR DESCRIPTION
**Pull request by dvargas92495**
This PR adds streaming support for TryNode's. It enables the p80 case of prompt nodes that have reject on error toggle on today.

This PR was blocking basic rag execution and has serialization implications as well

### CODE REVIEW

File: ee/vellum_ee/workflows/display/nodes/base_node_display.py
Comment (dvargasfuertes) [inline ee/vellum_ee/workflows/display/nodes/base_node_display.py]: would be nice to keep this API simple - especially since the param that was injected seemed like it could be handled at the consumer (see TryNodeDisplay below)

File: ee/vellum_ee/workflows/display/nodes/vellum/try_node.py
Comment (dvargasfuertes) [inline ee/vellum_ee/workflows/display/nodes/vellum/try_node.py]: Try nodes annotate their inner nodes. So, I think TryNodeDisplay should annotate its inner display, instead of passing error output id

File: ee/vellum_ee/workflows/display/nodes/vellum/try_node.py
Comment (dvargasfuertes) [inline ee/vellum_ee/workflows/display/nodes/vellum/try_node.py:39]: This is the main reason I wanted to pull error output id up - we are doing the same for definition

File: ee/vellum_ee/workflows/display/nodes/vellum/try_node.py
Comment (dvargasfuertes) [inline ee/vellum_ee/workflows/display/nodes/vellum/try_node.py]: More on this guy in `try_node.node` below

File: src/vellum/workflows/nodes/core/try_node/node.py
Comment (dvargasfuertes) [inline src/vellum/workflows/nodes/core/try_node/node.py:62]: rewrote this method to work more like subworkflow nodes, since we need to handle streaming

File: src/vellum/workflows/runner/runner.py
Comment (dvargasfuertes) [inline src/vellum/workflows/runner/runner.py:554]: thread names help with debugging

File: src/vellum/workflows/nodes/core/try_node/node.py
Comment (dvargasfuertes) [inline src/vellum/workflows/nodes/core/try_node/node.py]: This is the most important block in the PR to review

First some python context. When Python initiates a class in local scope, they use this special `<locals>` module path. We are borrowing from that pattern to introduce `<adornment>`.

Currently, we were emitting the same `WrappedNode` dynamic class back to vellum, so our translator had no idea how to map it back to a node id. Therefore, we needed a dynamic class so that our workflow translator in Vellum knows how to map it back to the PromptNode. Instead of making the class name dynamic, we are making the module path dynamic and downstream of the original wrapped node in order to distinguish which original node was using reject on error toggle.

I chose the word adornment for a few reasons:
- consistency with the UI
- differentiates from "decorator" - not all python decorators return a new class, let alone a new node

Eager for feedback here both on approach and choice

File: src/vellum/workflows/workflows/base.py
Comment (dvargasfuertes) [inline src/vellum/workflows/workflows/base.py:388]: new parent context require pydantic rebuilds of both events and bodies

File: tests/helpers.py
Comment (dvargasfuertes) [inline tests/helpers.py:10]: neat helper that assisted me in figuring out why a test was hanging. Thanks cursor

File: tests/workflows/stream_try_node_annotation/tests/test_workflow.py
Comment (dvargasfuertes) [inline tests/workflows/stream_try_node_annotation/tests/test_workflow.py:45]: Shows that we will see prompt node deltas in the UI

**Review by dvargas92495 — COMMENTED** [review]
CI is failing even though all of the tests pass locally. This PR is meaty, so I'm going to hope that resolving those tests will not result in huge diffs and request reviews now while I hunt down what's going on here

**Comment by dvargas92495** [comment]
omg I was on python 3.13 locally somehow - once I switched to python 3.9 I was able to repro the test failure. This means that this is only an issue for 3.9... fun -.-

**Comment by dvargas92495** [comment]
hell yea

File: src/vellum/workflows/nodes/core/try_node/node.py
Comment (noanflaherty) [inline src/vellum/workflows/nodes/core/try_node/node.py:138]: nit: A comment explaining what going on here would be helpful

File: src/vellum/workflows/nodes/displayable/tests/test_inline_text_prompt_node.py
Comment (noanflaherty) [inline src/vellum/workflows/nodes/displayable/tests/test_inline_text_prompt_node.py]: [oos] We should rename `WacVellumError`

File: tests/helpers.py
Comment (noanflaherty) [reply tests/helpers.py:10]: nit: a docstring would be helpful

File: ee/vellum_ee/workflows/display/nodes/vellum/try_node.py
Comment (noanflaherty) [reply ee/vellum_ee/workflows/display/nodes/vellum/try_node.py]: [4/5] I think we can get away with it in this very specific case due to the fact that
1) We happen to have inner nodes whose serialized repr all _happen_ to model this the same way, and
2) These nodes will go away at some point

However, more philosophically, I think the previous pattern is more true to the intention here. TryNode shouldn't have to know anything about what it wraps or the intended data model of the inner node's serialized representation. This is why we did it the way we did. I think of this more like django's serializer context – we're allowing access to parent serializer data in inner serializers. We do this now by passing a kwarg, but could do it just as well with a concept of context. In either case, I feel 4/5 that TryNode shouldn't know/care about what it wrapss

**Review by noanflaherty — APPROVED** [review]
Approving to unblock, but I'm not sold on the removal of `error_output_id` as a kwarg.

File: ee/vellum_ee/workflows/display/nodes/vellum/try_node.py
Comment (dvargasfuertes) [reply ee/vellum_ee/workflows/display/nodes/vellum/try_node.py]: I suppose error_output_id is node specific. definition isn't though - that will be universal. So proposal:
- some notion of context equivalent for error output id (or I'll just add back the kwarg if more than >15 min LoE)
- definition manipulation stays on the outside here

Comment (noanflaherty) [reply ee/vellum_ee/workflows/display/nodes/vellum/try_node.py]: agreed!

Comment (dvargasfuertes) [reply ee/vellum_ee/workflows/display/nodes/vellum/try_node.py]: I ended up adding back the kwarg - context on init doesn't work bc the error output id calculation needs the already initiated `node_display.node_id`.

Emotionally I still feel "an inner node doesn't need to know it's wrapped" more than "a wrapper node doesn't need to know what it's wrapping", particularly since the way the wrapper adds the `error` output itself is consistent across all nodes, but `error_output_id` itself should* be short lived anyway and defer to your domain DRI

Comment (dvargasfuertes) [reply ee/vellum_ee/workflows/display/nodes/vellum/try_node.py]: > TryNode shouldn't have to know anything about what it wraps or the intended data model of the inner node's serialized representation

I now disagree with this statement, also with a 4/5. The thing that a decorator wraps in python is defined as an input to the decorator. This means the decorator knows the type of the thing its wrapping and could discriminate and act accordingly. Not doing so is what I think is leading to a lot of adornment logic bleeding into base node components: https://github.com/vellum-ai/vellum-python-sdks/pull/655/files

**Pull request merged**
